File: scala/src/main/scala/bbc/dojonorth/lyrics/ReadTestData.scala
package bbc.dojonorth.lyrics

/**
 * Just to make sure the test data loads properly...
 * This main class takes the path to your mxm_dataset_test or mxm_dataset_train file
 */
object ReadTestData extends App {

  val loader = new LyricDataLoader

  if (args.length != 1) {
    println("Usage: ReadTestData <path to an MXM file>")
  }
  else {
    val songsData = loader.load(args(0))

    println("\nFirst five track IDs:")
    songsData.songs.take(5).foreach { song =>
      println(s"${song.trackId}")
    }

    println("\nFirst 5 words in words list:")
    songsData.words.take(5).foreach(println)
  }
}

File: javascript/lib/lyrics/ReadTestData.js
#!/usr/bin/env node

/**
 * Just to make sure the test data loads properly...
 * This main class takes the path to your mxm_dataset_test or mxm_dataset_train file
 */

var _ = require('lodash');
var LyricDataLoader = require('./LyricDataLoader');

var loader = new LyricDataLoader();
var args = process.argv;

if (args.length != 3) {
    console.log('Usage: ./ReadTestData.js <path to an MXM file>');
} else {
    console.time('load');
    songData = loader.load(args[2]);
    console.timeEnd('load');

    console.log();
    console.log('First five track IDs:');


    console.time('process songs');
    _.each(songData.songs.slice(0, 5), function (song) {console.log(song.trackId)});
    console.timeEnd('process songs');

    console.log();
    console.log('First five words in words list:');

    console.time('process words');
    _.each(songData.words.slice(0, 5), console.log);
    console.timeEnd('process words');
}
